src/io/flutter/utils/DiagnosticsNode.java

```diff
@@ -0,0 +1,502 @@
+   * will make the name self-evident.
+   */
+  public boolean getShowName() {
+    return getBooleanMember("showName", true);
+  }
+
+  /**
+   * Description to show if the node has no displayed properties or children.
+   */
+  public String getEmptyBodyDescription() {
+    return getStringMember("emptyBodyDescription");
+  }
+
+  /**
+   * Hint for how the node should be displayed.
+   */
+  public DiagnosticsTreeStyle getStyle() {
+    return getStyleMember("style", DiagnosticsTreeStyle.sparse);
+  }
+
+  /**
+   * Dart class defining the diagnostic node.
+   * For example, DiagnosticProperty<Color>, IntProperty, StringProperty, etc.
+   * This should rarely be required except for cases where custom rendering is desired
+   * of a specific Dart diagnostic class.
+   */
+  String getType() {
+    return getStringMember("type");
+  }
+
+  // The follow fields are only specified for properties.
+
+  /**
+   * Whether the description is enclosed in double quotes.
+   *
+   * Only relevant for String properties.
+   */
+  public boolean getIsQuoted() {
+    return getBooleanMember("quoted", false);
+  }
+
+  public boolean hasIsQuoted() {
+    return json.has("quoted");
+  }
+
+  /**
+   * Optional unit the [value] is measured in.
+   *
+   * Unit must be acceptable to display immediately after a number with no
+   * spaces. For example: 'physical pixels per logical pixel' should be a
+   * [tooltip] not a [unit].
+   * Only specified for Number properties.
+   */
+  public String getUnit() {
+    return getStringMember("unit");
+  }
+
+  public boolean hasUnit() {
+    return json.has("unit");
+  }
+
+  /**
+   * String describing just the numeric [value] without a unit suffix.
+   */
+  public String getNumberToString() {
+    return getStringMember("numberToString");
+  }
+
+  public boolean hasNumberToString() {
+    return json.has("numberToString");
+  }
+
+  /**
+   * Description to use if the property [value] is true.
+   *
+   * If not specified and [value] equals true the property's priority [level]
+   * will be [DiagnosticLevel.hidden].
+   *
+   * Only applies to Flag properties.
+   */
+  public String getIfTrue() {
+    return getStringMember("ifTrue");
+  }
+
+  public boolean hasIfTrue() {
+    return json.has("ifTrue");
+  }
+
+  /**
+   * Description to use if the property value is false.
+   *
+   * If not specified and [value] equals false, the property's priority [level]
+   * will be [DiagnosticLevel.hidden].
+   *
+   * Only applies to Flag properties.
+   */
+  public String getIfFalse() {
+    return getStringMember("ifFalse");
+  }
+
+  public boolean hasIfFalse() {
+    return json.has("ifFalse");
+  }
+
+  /**
+   * Value as a List of strings.
+   *
+   * The raw value can always be extracted with the regular observatory protocol.
+   *
+   * Only applies to IterableProperty.
+   */
+  public ArrayList<String> getValues() {
+    ArrayList<String> values = null;
```

Comment:
probably don't need to init this

Reply:
I get a "values" might not have been initialized java lint warning otherwise.
Cleaned up the method to use an early return style to reduce nesting and avoid the = null.

Reply:
odd